lib/filesystem/include/irods/filesystem/path_utilities.hpp

```diff
@@ -0,0 +1,46 @@
+#ifndef IRODS_FILESYSTEM_PATH_UTILITIES_HPP
+#define IRODS_FILESYSTEM_PATH_UTILITIES_HPP
+
+#include "irods/filesystem/path.hpp"
+#include "irods/filesystem/filesystem_error.hpp"
+
+#include "irods/rodsDef.h"
+#include "irods/rodsErrorTable.h"
+#include "irods/system_error.hpp"
+
+#include <cstring>
+
+namespace irods::experimental::filesystem
+{
+    /// Returns whether the path element \p _c represents the preferred path separator.
+    ///
+    /// 4.2.12
+    inline auto is_separator(path::value_type _c) noexcept -> bool
+    {
+        return path::preferred_separator == _c;
+    }
+
+    /// Throws a filesystem_error if the length of path \p _p exceeds the maximum length
+    /// allowed for a logical path.
+    ///
+    /// 4.2.12
+    inline auto throw_if_path_length_exceeds_limit(const path& _p) -> void
+    {
+        if (std::strlen(_p.c_str()) > MAX_NAME_LEN) {
```

Comment:
Codacy is correct. Not only that, but the operator should be `>=` because `MAX_NAME_LEN` represents the buffer size including a null terminating byte, but `std::strlen` does not.

Reply:
Replaced `std::strlen` with `strnlen` to account for the fixed size.